openhtf/util/units/bin/units_from_xls.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Read in a .xls file and generate a units module for OpenHTF.
+
+UNECE, the United Nations Economic Commision for Europe, publishes a set of
+unit codes for international trade in the form of Excel spreadsheets (.xls
+files). Various revisions of the spreadsheet can be found as part of the the
+downloadable "Codes for Units of Measurement used in the International Trade"
+.zip archive listed here:
+
+http://www.unece.org/cefact/codesfortrade/codes_index.html
+
+This tool is used to parse those spreadsheets and turn the published standard
+code set into a sub-module inside OpenHTF.
+
+Typical usage of this generation script looks like:
+
+python ./bin/units_from_xls.py --infile ~/rec20_Rev9e_2014.xls \
+--outfile ./__init__.py
+
+Legal python names are generated for the UnitInfo objects from the "Name" field
+on the spreadsheet, generally by removing special characters, turning spaces
+into underscores, and coverting to uppercase.
+"""
+
+
+import argparse
+import os
+import shutil
+import re
+import sys
+from tempfile import mkstemp
+
+import xlrd
+
+
+# Column names for the columns we care about. This list must be populated in
+# the expected order: [<name label>, <code label>, <suffix label>].
+COLUMN_NAMES = ['Name',
+                'Common\nCode',
+                'Symbol']
+
+PRE = '''# coding: iso-8859-1
```

Comment:
nit: this coding comment may not be accurate since you `string.encode('utf8')` below.